### PR TITLE
fix(providers): update providers config from verification walk-throughs

### PR DESCRIPTION
**Pull request by ndom91**
## ☕️ Reasoning

- Update 3 providers based on findings from our verification of the example app providers

1. **Netlify** - Simply append `?scope` to the `authorizationUrl`
2. **Slack** - Set `checks: "nonce"`
3. **WorkOS** - Add `connection` argument / env var which gets appended to `authorziationUrl` as a search param

@balazsorban44 two question:
1. I saw in the types that `checks: ("none" | "state" | "pkce")`, are the types wrong or is `nonce` really an invalid value?
2. Regarding WorkOS, I wasn't sure how best to pass this unique value, `AUTH_WORKOS_CONNECTION`. Using `process.env` directly in the provider seems wrong since that won't work in all environments.. Do you have a better idea for this one? :thinking: 

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

**Review by balazsorban44 — APPROVED** [review]

File: docs/pages/getting-started/providers/workos.mdx [hunk 1]
```diff
@@ -35,7 +35,7 @@ https://example.com/auth/callback/workos
 ```
 AUTH_WORKOS_ID
 AUTH_WORKOS_SECRET
-AUTH_WORKOS_ISSUER
+AUTH_WORKOS_CONNECTION
```
Comment (balazsorban44) [inline docs/pages/getting-started/providers/workos.mdx:38]: Let's just ask the user to pass `connection` as a property.

File: packages/core/src/providers/workos.ts [hunk 2]
```diff
@@ -144,15 +144,20 @@ export interface WorkOSProfile extends Record<string, any> {
  * :::
  */
 export default function WorkOS<P extends WorkOSProfile>(
-  options: OAuthUserConfig<P>
+  options: OAuthUserConfig<P> & { connection?: string }
 ): OAuthConfig<P> {
-  const { issuer = "https://api.workos.com/" } = options
+  const {
+    issuer = "https://api.workos.com/",
+    connection = process.env.AUTH_WORKOS_CONNECTION ?? "",
```
Comment (balazsorban44) [inline packages/core/src/providers/workos.ts:151]: we cannot use `process.env` in core as it's non-standard. let's just ask the user to pass `connection` for now.

**Review by balazsorban44 — CHANGES_REQUESTED** [review]

**Comment by balazsorban44** [comment]
Can you also modify the [`providers` list](https://github.com/nextauthjs/next-auth/blob/599bdf2e2dde1cee9b4ae53ca1d3a7996bb9cb09/apps/examples/nextjs/auth.ts#L34) accordingly?

File: packages/core/src/providers/workos.ts (hunk 2 above)
Comment (ndom91) [reply packages/core/src/providers/workos.ts:151]: Yeah thats what i wrote in the description as well. Sounds good, i'll just change it to take a connection option then :+1:

Comment (ndom91) [reply packages/core/src/providers/workos.ts:151]: Also the other question I had in the description - for `checks` we had an enum including the option `none`, but not `nonce`. Is `nonce` correct? Should I update the `checks` enum?

**Review by balazsorban44 — APPROVED** [review]

**Pull request merged**